pkg/controller/controller.go

```diff
@@ -595,9 +595,24 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		return nil, controller.ProvisioningFinished, err
 	}
 
+	pvcAnnotationsForSCParam := map[string]string{}
+	for k, v := range claim.GetAnnotations() {
+		pvcAnnotationsForSCParam["pvc.annotations['"+k+"']"] = v
+	}
+
 	fsTypesFound := 0
 	fsType := ""
 	for k, v := range sc.Parameters {
+		// Replace SC parameters with PVC annoations
+		if strings.Contains(v, "pvc.annotations") {
+			resolvedName, err := resolveTemplate(v, pvcAnnotationsForSCParam)
```

Comment:
write like this:
```
if resolvedName, err := resolveTemplate(v, pvcAnnotationsForSCParam); err != nil {

}
```

Reply:
If we write it like this, variable resolvedName can't be reached out of "if {}" scope (line 614)